### ISSUE TITLE
What does it mean? Cannot swizzle selector 'application:didReceiveRemoteNotification:fetchCompletionHandler:' of class 'AppDelegate'...

### ISSUE THREAD

**Issue by biapar** [Microsoft/mobile-center-sdk-dotnet#438]
[MobileCenter] ERROR: +[MSAppDelegateForwarder swizzleOriginalSelector:withCustomSelector:originalClass:]_block_invoke/207 Cannot swizzle selector 'application:didReceiveRemoteNotification:fetchCompletionHandler:' of class 'AppDelegate'. You will have to explicitly call APIs from Mobile Center in your app delegate implementation.

**Comment by clpolet** [comment]
The `MSAppDelegateForwarder` is the component of the iOS Mobile Center SDK responsible for forwarding calls for some methods of the `UIApplicationDelegate` to the Mobile Center SDK and your application delegate. The current version of the MC SDK has a limitation on the `application:didReceiveRemoteNotification:fetchCompletionHandler:` method which is `void DidReceiveRemoteNotification(UIApplication application, NSDictionary userInfo, System.Action<UIBackgroundFetchResult> completionHandler` in the Xamarin world. the MC SDK can't set up the forwarding on this particular method if it has already been implemented by your application delegate. 

Basically if you override the method `void DidReceiveRemoteNotification(UIApplication application, NSDictionary userInfo, System.Action<UIBackgroundFetchResult> completionHandler` in your  `AppDelegate` class then you'll have to do the forwarding yourself by calling `Push.DidReceiveRemoteNotification(userInfo)`.

There is an example in the [documentation see "4. Implement the callback to enable push event
"](https://docs.microsoft.com/en-us/mobile-center/sdk/push/xamarin-ios#disable-automatic-forwarding-of-application-delegates-methods-to-mobile-center-services).

This limitation is going to be fixed in a later version of the SDK.

**Comment by guperrot** [comment]
Closing until further feedbacks.
